core/src/main/java/org/apache/calcite/adapter/clone/ArrayTable.java

```diff
@@ -243,9 +242,9 @@ public static List asList(final Representation representation,
 
     /** Converts a value set into a compact representation. If
      * {@code sources} is not null, permutes. */
-    Object freeze(ColumnLoader.ValueSet valueSet, int[] sources);
+    Object freeze(ColumnLoader.ValueSet valueSet, int @Nullable [] sources);
```

Comment:
It seems some annotations appear before the type, while others appear after the type. We should make this consistent?

Reply:
Please read https://docs.oracle.com/javase/specs/jls/se8/html/jls-9.html#jls-9.7.4 amd 
https://github.com/vlsi/calcite/blob/checkerframework/site/develop/index.md#null-safety

`@Nullable int[]` and `int @Nullable []` are different.